fastlane/Fastfile

```diff
@@ -50,6 +54,31 @@ platform :ios do
     replace_ios_assets()
   end
 
+  lane :build_qa do
+    match(type: 'adhoc')
+    build_ios
+
+    current_build_number = get_build_number(xcodeproj: './ios/Mattermost.xcodeproj').to_i
+    plist_template = File.read('plist.erb')
+    abbreviated_commit_hash = `git rev-parse --short head`
+    abbreviated_commit_hash.strip!
+    plist_body = ERB.new(plist_template).result(binding)
+
+    s3 = Aws::S3::Resource.new(region: ENV['AWS_REGION'])
+    plist_obj = s3.bucket(ENV['AWS_BUCKET_NAME']).object("ios/#{abbreviated_commit_hash}.plist")
+    plist_obj.put(body: plist_body)
+
+    ipa_obj = s3.bucket(ENV['AWS_BUCKET_NAME']).object("ios/#{abbreviated_commit_hash}.ipa")
```

Comment:
What if ENV['AWS_REGION']  is null or not set, same goes with `AWS_BUCKET_NAME`, should we have a default to fallback or something?

Reply:
Currently we're not supporting anything except AWS but I think as a new feature we could support uploading directly to the channel (simply, for Android) and another upload strategy for iOS. The reason iOS can't use a simple file upload is that we need to upload 2 files with the `info.plist` able to know the path to the `ipa`. Can it be done other ways: absolutely but it's not currently a feature.

Reply:
I get that but what im saying is what happens if I run the script and the env var is not set

Reply:
Oh you're proposing giving users a nicely printed message that it's not set?

Reply:
either that or fallback to a default value

Reply:
I guess is fine for now, we can add that later if we plan to share this script with others